### DISCUSSION TITLE
代码生成时如何生成一个一级菜单

### DISCUSSION THREAD

**Discussion by hunao0221** [zuihou/zuihou-admin-cloud#303] [企业版]
<img width="1379" alt="image" src="https://github.com/dromara/lamp-cloud/assets/11972530/8a0e58fb-1659-45e9-9607-33ffef96e6ed">

如图所示，这里必须选择一个已存在的菜单，那么生成的菜单至少是二级的了。
当前有办法实现生成一级菜单吗？还是只能修改代码加入parent_id = 0的选择?

**Comment by zuihou** [comment]
不能。 

就算这里选择了菜单，也不会直接执行SQL语句，可以在自己执行时， 修改SQL语句中的parent_id

**Comment by zuihou** [comment]
<img width="1375" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/8d703d27-818d-4c73-a520-368967b85120">
改成0后，自己执行

**Reply by hunao0221** [reply]
tree_path也需要修改！

**Reply by zuihou** [reply]
tree_path, tree_grade

**Comment by zuihou** [comment]
后面可以做个优化， 上级菜单 改成非必填， 没选就默认是1级菜单
